### PR TITLE
feat(truncate collections): introduce

### PR DESCRIPTION
**Pull request by miklos-martin**
@blacksonic please take a look at this, thank you

### CODE REVIEW

File: lib/truncate-collections/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+  get _collections() {
+    return this._mongoose.connection.collections;
+  }
+
+}
+
+module.exports = function(mongoose) {
+  return (new TruncateCollection(mongoose)).truncate;
```
Comment (Wilzi) [inline lib/truncate-collections/index.js]: please use multiline (cleancode)

File: lib/truncate-collections/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      .map((collection) => this._truncateCollection(collection));
+
+    return Promise.all(truncateCollections);
+  }
+
+
+  _truncateCollection(collection) {
+    return new Promise(function (resolve, reject) {
+      collection.remove({}, function (err) {
+        if (err && err.message !== 'ns not found') reject(err);
```
Comment (Wilzi) [inline lib/truncate-collections/index.js]: do not use single line if statement

File: lib/truncate-collections/index.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+class TruncateCollection {
+
+  constructor(mongoose) {
+    this._mongoose = mongoose;
+    this.truncate = this.truncate.bind(this);
+  }
+
+
+  truncate() {
+    let truncateCollections = this._collectionNames
```
Comment (Wilzi) [inline lib/truncate-collections/index.js]: you should use const instead of let

File: lib/truncate-collections/index.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+class TruncateCollection {
+
+  constructor(mongoose) {
+    this._mongoose = mongoose;
+    this.truncate = this.truncate.bind(this);
```
Comment (peter-vasarhelyi) [inline lib/truncate-collections/index.js]: why do you bind truncate here? you could do it in the exported function.

File: lib/truncate-collections/index.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+
+  truncate() {
+    let truncateCollections = this._collectionNames
+      .map((collectionName) => this._collections[collectionName])
```
Comment (Wilzi) [inline lib/truncate-collections/index.js]: you should delete the brackets for single input parameter (linter should throw an error about this)

**Review by Wilzi — CHANGES_REQUESTED** [review]
missing test

File: lib/truncate-collections/index.js (hunk 4 above)
Comment (miklos-martin) [reply lib/truncate-collections/index.js]: it makes the exported function a one-liner

File: lib/truncate-collections/.#index.js [hunk 6]
```diff
@@ -0,0 +1 @@
+mmartin@mmartin-mbptb.local.24940
```
Comment (Wilzi) [inline lib/truncate-collections/.#index.js]: ? :)

**Review by Wilzi — CHANGES_REQUESTED** [review]

File: lib/truncate-collections/.#index.js (hunk 6 above)
Comment (miklos-martin) [reply lib/truncate-collections/.#index.js]: wow, sorry

File: lib/truncate-collections/index.js (hunk 4 above)
Comment (peter-vasarhelyi) [reply lib/truncate-collections/index.js]: it is much more important to have an easily readable code, than a one liner function :)

**Comment by miklos-martin** [comment]
I've squashed the commits with the fixes

**Comment by Wilzi** [comment]
LGTM

**Pull request merged**